lib/cartopy/tests/mpl/test_gridliner.py

```diff
@@ -238,8 +209,8 @@ def test_grid_labels():
 
 
 @pytest.mark.skipif(
-    MPL_VERSION < '2.0.0',
-    reason='Impossible to override tight layout algorithm in mpl < 2.0.0')
+    MPL_VERSION < '3.0.0',
```

Comment:
I don't follow. The old code used to run on Matplotlib >=2...why do we need to update this rather than flat out remove?

Reply:
Great question! I was wondering the same thing. This is a new test added in the last month or so, and I think our previous CI was only running MPL 1.x and MPL 3.x, never hitting the 2.x builds from what I can see in the Travis setup.

@lukelbd do you know if this test should be possible to run in MPL 2.x? I haven't looked into the tight_layout code on MPLs side at all to know when it was changed and what should be expected.

Reply:
This change should remain -- I'm not sure why, but turns out the tight layout test will not run successfully until matplotlib >= 3.0. Originally I thought it should work in matplotlib >= 2.0 but [my reasoning was incorrect](https://github.com/SciTools/cartopy/pull/1355#issuecomment-643846634).

After some empirical testing with:

```sh
conda create -c conda-forge/label/testing -c conda-forge -n cartopy-dev-testing cython numpy matplotlib-base=VERSION proj4 pykdtree scipy fiona pillow pytest pytest-xdist filelock pep8 pyshp shapely six requests pyepsg owslib
pip install -e .
echo 'backend: agg' > matplotlibrc
conda activate cartopy-dev-testing
pytest -k 'grid_labels_tight'
```

the last matplotlib-base 2.X version available on conda fails, and the first matplotlib-base 3.X version works.

Reply:
Thanks for the quick response!

I think this is fine for now then, since it is a new feature and we aren't breaking anything that was possible previously. It actually just defaults to the previous 0.18 behavior.